extension-manifest-v2/app/content-scripts/renew.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import { showIframe } from '@ghostery/ui/iframe';
+
+const HOUR_IN_MS = 1000 * 60 * 60;
+
+chrome.runtime.sendMessage({ action: 'renew:setup' }).then((renew_setup) => {
```

Comment:
can we make it a push instead of a pull? additional message on every page load for sake of one time event is not great.

Reply:
Unfortunately, I think we cannot. This works the same as never-consent, so...

If we would "push" a message from the background, who should be the receiver? Active tab? It can be a settings page or another from the browser (so there is no required content script). If we filter them out, then it can be a tab that is not active anymore...

If you now some trick, let me know ;)

I think I already explained that here: https://github.com/ghostery/ghostery-extension/pull/1205#discussion_r1237122111

Reply:
I can see would be to make a combination of mechanisms like follow:
* move triggering to the background - it will by time bound
* on trigger, create a new webNavigation listener that await a capable website (with URL that looks like a real page)
* on page found, send a message to the content script after X (2?) seconds 

The supposition is to show messages quickly after a page load.

Reply:
As this is MV2, we can add listener later, so i think it might work. I'll try to do it, and get it out.